core/translate/group_by.rs

```diff
@@ -156,14 +163,22 @@ pub fn emit_group_by<'a>(
 
     let group_by = plan.group_by.as_ref().unwrap();
 
+    let non_aggregate_count = plan
+        .result_columns
+        .iter()
+        .filter(|rc| !rc.contains_aggregates)
+        .count();
+
+    let agg_args_count = plan
+        .aggregates
+        .iter()
+        .map(|agg| agg.args.len())
+        .sum::<usize>();
+
     // all group by columns and all arguments of agg functions are in the sorter.
     // the sort keys are the group by columns (the aggregation within groups is done based on how long the sort keys remain the same)
-    let sorter_column_count = group_by.exprs.len()
-        + plan
-            .aggregates
-            .iter()
-            .map(|agg| agg.args.len())
-            .sum::<usize>();
+    let sorter_column_count = non_aggregate_count + agg_args_count;
```

Comment:
The comment above should now say "all non-aggregate columns and all arguments of agg functions are in the sorter"